### DISCUSSION TITLE
using `addMocksToSchema`'s `mocks` prop no longer works for arrays

### DISCUSSION THREAD

**Discussion by RadomirNowak** [dotansimha/graphql-code-generator#8754] [Q&A]
Hi
Recently I've been trying to upgrade a few graphql-codegen libraries. I've been using `addMockFunctionsToSchema` (now `addMocksToSchema`) to setup mocks for each cypress test that I'm running like so:
```
const mockedResponses = {
    Query: () => ({
        getDocument: () => ({
            id: 'mocked-id', // this is returned in call response
            tags: [ // this is `null` is call response
                { id: 'tag-one' }
            ],
        })
    })
}

const schemaWithMocks = addMocksToSchema({ schema, mocks: mockedResponses });
const client = new ApolloClient({
    link: new SchemaLink({ schema: schemaWithMocks }),
    cache: new InMemoryCache(),
});

...
```

After upgrading and making quite a lot changes it seems like now the returned mocked data does not work with arrays. As a response to my graphql query I get all the mocked values except for the `tags` property which is null. If I remove the property altogether, I get randomly generated values. What I would like is to get same data returned to me and only fields that are not specified to be randomly generated.

It seems like one of the versions between my old one and new one has broken the functionality of `mocks` property.

Can you point me how can I debug it further to understand if there's something that I need to change or is it maybe a bug in the library itself?

Thanks

Below are packages that I upgraded:
From:
```
    "@apollo/client": "3.4.10",
    "@graphql-codegen/add": "3.2.1",
    "@graphql-codegen/cli": "1.11.2",
    "@graphql-codegen/introspection": "1.11.2",
    "@graphql-codegen/typescript": "1.11.2",
    "@graphql-codegen/typescript-operations": "1.11.2",
    "@graphql-codegen/typescript-react-apollo": "1.11.2",
    "@graphql-codegen/typescript-resolvers": "2.7.7",
    "@graphql-tools/wrap": "6.2.4",
    "graphql": "14.7.0",
    "graphql-playground-react": "1.7.21",
```
To:
```
"@apollo/client": "3.7.2",
    "@graphql-codegen/add": "3.2.3",
    "@graphql-codegen/cli": "2.16.1",
    "@graphql-codegen/introspection": "2.2.3",
    "@graphql-codegen/typescript": "2.8.5",
    "@graphql-codegen/typescript-operations": "2.5.10",
    "@graphql-codegen/typescript-react-apollo": "3.3.7",
    "@graphql-codegen/typescript-resolvers": "2.7.10",
    "@graphql-tools/mock": "8.7.14",
    "@graphql-tools/schema": "9.0.12",
    "@graphql-tools/utils": "9.1.3",
    "@graphql-tools/wrap": "9.2.22",
    "graphql": "16.6.0",
    "graphql-playground-react": "1.7.28",
```

**Comment by ardatan** [comment]
Please create an issue with a reproduction by following our bug report template;
https://github.com/dotansimha/graphql-code-generator/issues/new?assignees=&labels=&template=bug_report.yml
